MOM6/patches/mom_cap.F90.patch

```diff
@@ -1,6 +1,271 @@
+   type(ice_ocean_boundary_type), pointer :: ice_ocean_boundary_type_ptr
++#ifdef _USE_GENERIC_TRACER
++  type(coupler_2d_bc_type),      pointer :: coupler_2d_bc_type_ptr
++#endif
+ end type
+ 
+ !>  Wrapper-derived type required to associate an internal state instance
+@@ -408,6 +422,10 @@ subroutine InitializeAdvertise(gcomp, importState, exportState, clock, rc)
+   type (ocean_public_type),      pointer :: ocean_public => NULL()
+   type (ocean_state_type),       pointer :: ocean_state => NULL()
+   type(ice_ocean_boundary_type), pointer :: Ice_ocean_boundary => NULL()
++  type(coupler_1d_bc_type),      pointer :: gas_fields_atm => NULL()
++  type(coupler_1d_bc_type),      pointer :: gas_fields_ocn => NULL()
++  type(coupler_1d_bc_type),      pointer :: gas_fluxes => NULL()
++  type(coupler_2d_bc_type),      pointer :: atm_fields => NULL()
+   type(ocean_internalstate_wrapper)      :: ocean_internalstate
+   type(ocean_grid_type),         pointer :: ocean_grid => NULL()
+   type(directories)                      :: dirs
+@@ -439,6 +457,7 @@ subroutine InitializeAdvertise(gcomp, importState, exportState, clock, rc)
+   character(len=512)                     :: restartfile          ! Path/Name of restart file
+   character(len=2048)                    :: restartfiles         ! Path/Name of restart files
+                                                                  ! (same as restartfile if single restart file)
++  character(240)                         :: additional_restart_dir
+   character(len=*), parameter            :: subname='(MOM_cap:InitializeAdvertise)'
+   character(len=32)                      :: calendar
+   character(len=:), allocatable          :: rpointer_filename
+@@ -517,6 +536,8 @@ subroutine InitializeAdvertise(gcomp, importState, exportState, clock, rc)
+   if (chkerr(rc,__LINE__,u_FILE_u)) return
+   call MOM_infra_init(mpi_comm_mom)
+ 
++  call field_manager_init
++
+   ! determine the calendar
+   if (cesm_coupled) then
+     call NUOPC_CompAttributeGet(gcomp, name="calendar", value=cvalue, &
+@@ -648,13 +669,44 @@ subroutine InitializeAdvertise(gcomp, importState, exportState, clock, rc)
+ 
+   endif
+ 
++  ! Set NUOPC attribute additional_restart_dir to RESTART/ if not defined
++  additional_restart_dir = "RESTART/"
++  call NUOPC_CompAttributeGet(gcomp, name="additional_restart_dir", value=cvalue, &
++        isPresent=isPresent, isSet=isSet, rc=rc)
++  if (ChkErr(rc,__LINE__,u_FILE_u)) return
++  if (isPresent .and. isSet) then
++    additional_restart_dir = slasher(cvalue)
++  else
++    call ESMF_LogWrite('MOM_cap:additional_restart_dir unset. Defaulting to '//trim(additional_restart_dir), &
++          ESMF_LOGMSG_INFO)
++  endif
++  call NUOPC_CompAttributeSet(gcomp, name="additional_restart_dir", value=additional_restart_dir, rc=rc)
++  if (chkerr(rc,__LINE__,u_FILE_u)) return
++
+   ocean_public%is_ocean_pe = .true.
++#ifdef _USE_GENERIC_TRACER
++  ! Initialise structures for extra tracer fluxes
```

Comment:
Would there be a way of only calling the generic tracers code if needed? Probably the memory and CPU footprint of these calls is small, but still, would be nice to skip them if no generic tracers are needed.

Reply:
Possibly, but I'll have to think about that. I haven't looked at the memory, but I did some [basic tests](https://github.com/COSIMA/access-om3/issues/166) that show performance isn't degraded with this code is included but no generic tracers are configured